api/v1/clustercatalog_types.go

```diff
@@ -131,8 +133,8 @@ type ClusterCatalogSpec struct {
 	// The highest possible value is 2147483647.
 	//
 	// +kubebuilder:default:=0
-	// +kubebuilder:validation:minimum:=-2147483648
-	// +kubebuilder:validation:maximum:=2147483647
+	// +kubebuilder:validation:Minimum:=-2147483648
+	// +kubebuilder:validation:Maximum:=2147483647
```

Comment:
This was a typo that meant this was not properly being enforced. This is now a breaking change as these APIs have already been released as v1.

I'll leave it up to the maintainers as to how this should be handled. Some options I can quickly think of:
-  make the change, expect ratcheting validations to take over where used in production. Seems unlikely that users could have reasonably specified something out of these bounds.
- don't make the change. Leave unvalidated.

Reply:
IHMO, we should make this change.
**It is a bug,** not a breaking change.
We are not validating something that we should validate.

Reply:
Just because it is a bug does not make it a non-breaking change. This can be both, but it is most certainly a "breaking change" in the sense that you are adding restrictions to something that previously had no restrictions.

Reply:
I think this is a safe enough change. The GoDoc documentation correctly states our intent on the min and max values.